### ISSUE TITLE
Add support for normalization field in Collation

### ISSUE THREAD

**Issue by jyemin** [go-mgo/mgo#308]
MongoDB 3.4 collation support includes a "normalization" field of type boolean.  It should be added to the new Collation struct.

**Comment by niemeyer** [comment]
Yeah, as mentioned I expected that given it was in the actual index, despite not being in the design doc.

No problems, though. Will be on the next stable.

**Comment by jyemin** [comment]
Also required is a maxVariable field of type string.

**Comment by jyemin** [comment]
maxVariable: Controls which characters are affected by alternate: “shifted”.

   “punct” (punctuation and spaces are affected)
   “space” (only spaces)

    Default is "punct"


normalization: If true, normalizes text into Unicode NFD. 

   Default is false

**Comment by niemeyer** [comment]
Thanks for the updates, @jyemin.
